lib/src/formatter.dart

```diff
@@ -69,9 +69,10 @@ class DetailedReporter extends SimpleFormatter {
     var contents = getLineContents(line, error);
     out.writeln(contents);
 
-    var arrows = '^' * error.length;
     var spaces = column - 1;
-    var result = '${" " * spaces}$arrows';
+    var arrows = min(error.length, contents.length - spaces);
```

Comment:
Consider changing to:

var arrows = max(1, min(error.length, contents.length - spaces));

just to make sure at least one arrow is always printed.